### PR TITLE
Update and extend NO_BACKSLASH_ESCAPES implementation

### PR DESCRIPTION
**Pull request by lauxjpn**
Fully implement `NO_BACKSLASH_ESCAPES`, based on the new support from MySqlConnector.
https://github.com/mysql-net/MySqlConnector/issues/701
#827

Add `SetSqlModeOnOpen` option, to automatically execute a `set sql_mode = ...` operation, right after a connection has been opened.

Add `setSqlModeOnOpen = true` parameter to `DisableBackslashEscaping()`, to make usage of this option more intuitive. Can be set to `null`, to not change the current `SetSqlModeOnOpen` value.

Add/update test cases, fixtures etc.

When this goes GA, the wiki regarding `NO_BACKSLASH_ESCAPES` needs to be updated.

Current test metrics:

MySQL | Total | Passed | Failed | Skipped
------|-------|--------|--------|--------
8.0.17 | 14114 | 13487 | 148 | 479
5.7.27 | 14106 | 13436 | 183 | 487

### CODE REVIEW

File: src/EFCore.MySql/Infrastructure/Internal/MySqlOptionsExtension.cs [hunk 1]
```diff
@@ -73,6 +73,12 @@ protected override RelationalOptionsExtension Clone()
         /// </summary>
         public bool NoBackslashEscapes { get; private set; }
 
+        /// <summary>
+        ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        public bool UpdateSqlModeOnOpen { get; private set; }
```
Comment (bgrainger) [inline src/EFCore.MySql/Infrastructure/Internal/MySqlOptionsExtension.cs]: This needs to be added to the `MySqlOptionsExtension([NotNull] MySqlOptionsExtension copyFrom)` constructor (that clones the instance).

Comment (lauxjpn) [reply src/EFCore.MySql/Infrastructure/Internal/MySqlOptionsExtension.cs]: Good catch!

File: src/EFCore.MySql/Infrastructure/MySqlDbContextOptionsBuilder.cs [hunk 2]
```diff
@@ -66,8 +66,23 @@ public virtual MySqlDbContextOptionsBuilder EnableRetryOnFailure(int maxRetryCou
...
+            if (setSqlModeOnOpen.HasValue)
+            {
+                builder = builder.WithOption(e => e.SetSqlModeOnOpen(setSqlModeOnOpen.Value));
+            }
+
+            return builder;
+        }
+
+        /// <summary>
+        ///     Configures the context to use the default retrying <see cref="IExecutionStrategy" />.
```
Comment (bgrainger) [inline src/EFCore.MySql/Infrastructure/MySqlDbContextOptionsBuilder.cs]: Incorrect description in comment.

File: src/EFCore.MySql/Infrastructure/MySqlDbContextOptionsBuilder.cs [hunk 3]
```diff
@@ -66,8 +66,23 @@ public virtual MySqlDbContextOptionsBuilder EnableRetryOnFailure(int maxRetryCou
         ///     This applies to both constant and parameter values (i. e. user input, potentially).
         ///     Use this option if SQL mode NO_BACKSLASH_ESCAPES is guaranteed to be active.
         /// </summary>
-        public virtual MySqlDbContextOptionsBuilder DisableBackslashEscaping() =>
-            WithOption(e => e.DisableBackslashEscaping());
+        public virtual MySqlDbContextOptionsBuilder DisableBackslashEscaping(bool? setSqlModeOnOpen = true)
```
Comment (bgrainger) [inline src/EFCore.MySql/Infrastructure/MySqlDbContextOptionsBuilder.cs]: Need to document `setSqlModeOnOpen`?

**Pull request merged**